### PR TITLE
update training doc on platform services

### PR DESCRIPTION
**Pull request by justinpjose**
### What
**Describe what you have changed and why.**
- Added a new markdown which informs about the different platform services that we use
- Added images so that people naturally become familiar with the UI and understand what the outcome should be when accessing these services

### How to review
**Describe the steps required to test the changes.**
- View the `PLATFORM.md` file [here](https://github.com/ONSdigital/dp/blob/ee9c36cf1fd7dbe7fef2a0a4e592f7e33b9b67fe/training/platform-services/PLATFORM.md)
- Check if the content is clear, valid, and sufficient

### Who can review
**Describe who worked on the changes, so that other people can review.**
Anyone - Justin made the changes

### CODE REVIEW

File: training/platform-services/PLATFORM.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+Grafana informs how a process is behaving on the machine (its use of CPU, memory, use of IO operations, etc). In Grafana, the [Nomad Task][grafana-dev-nomad-tasks] dashboard is commonly used, which gives a list of tasks (each related to a specific application). From there, you can see the behaviours of specific applications to the machine such as memory usage (e.g. to identify a potential memory leak by changing the range of time to 7 days and analysing the graph). Grafana assists in debugging and we can look at [Kibana](#kibana) to see logs at a particular time based on any anomalies discovered from graphs in Grafana. Furthermore, the [Node Stats 2][grafana-dev-node-stats-2] dashboard is used as well which is specific to EC2 instances performance.
+
+Please note that CPU can spike as reserved CPU can exceed the limit because of its soft limit (CPU bursting). On the other hand, memory usage cannot spike as a hard limit is set for reserved memory and if memory exceeds its limit, the service will stop and restart
```
Comment (eldeal) [inline training/platform-services/PLATFORM.md:129]: Technically, they can both spike, in that a spike it a rapid change in use. The issue is whether that use exceeds limits. So maybe this could be better phrased around 'CPU can exceed its defined limits (known as bursting), as they are 'soft limits' Memory has a 'hard limit' and any use in excess will cause the service to stop/restart' or something?

File: training/platform-services/PLATFORM.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+More information about Grafana can be found on their website - https://grafana.com/docs/grafana/latest/
+
+## Prometheus
+Visit the following to access Prometheus
+- DEVELOP:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;https://prometheus.develop.onsdigital.co.uk
+- PRODUCTION:&nbsp;&nbsp;&nbsp;https://prometheus.production.onsdigital.co.uk
+
+[<img src="images/prometheus-dev-home.png" title="Prometheus Develop Home" width=49.7% height=49.7%/>][prometheus-dev-home] [<img src="images/prometheus-dev-alerts.png" title="Prometheus Develop Alerts" width=49.7% height=49.7%/>][prometheus-dev-alerts]
+
+Prometheus provides alerts and alarms to notify us on technical issues. These alerts set in Prometheus is alarmed in Slack. Prometheus stores all data as time series in which a time series graph can be viewed to report information. Therefore, more details about the alert can be retrieved from Prometheus. Prometheus uses the flexible query language [PromQL][prometheus-querying] to let us select and aggregate time series data in real time, speeding up the process of debugging
```
Comment (eldeal) [inline training/platform-services/PLATFORM.md:140]: > These alerts set in Prometheus is alarmed in Slack.

probably 'These alerts set in Prometheus trigger alarm messages in Slack.' ?

Comment (eldeal) [inline training/platform-services/PLATFORM.md:140]: > Prometheus stores all data as time series in which a time series graph can be viewed to report information. Therefore, more details about the alert can be retrieved from Prometheus. 

I feel like this isn't really telling me a lot, but can't put my finger on what to say instead. Maybe something like 'Prometheus stores all data as a time series, and monitors for changes over time. The information can be viewed in prometheus in a chart, and these charts are linked to from alerts so more detail is available as you investigate.' Does that make sense and still convey the same things? Sometimes I find in trying to define things, we use the same word too much and it gets hard to follow a bit (but that might just be me!)

File: training/platform-services/PLATFORM.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+   [nomad-scheduling]: <https://www.nomadproject.io/docs/internals/scheduling/scheduling>
+   [kibana-dev-home]: <https://kibana.develop.onsdigital.co.uk>
+   [helper-script-dp-setup]: <https://github.com/ONSdigital/dp-setup/tree/develop/scripts>
+   [dp-configs-secrets]: <https://github.com/ONSdigital/dp-configs/tree/master/secrets>
+   [dp-setup-vault-policies]: <https://github.com/ONSdigital/dp-setup/tree/develop/ansible/files/vault-policies>
+   [vault-policy]: <https://www.vaultproject.io/docs/concepts/policies>
+   [florence-vault-policy]: <https://github.com/ONSdigital/dp-setup/blob/develop/ansible/files/vault-policies/florence.hcl>
+   [vault-develop]: <https://vault.develop.onsdigital.co.uk>
+   [grafana-dev-home]: <https://grafana.develop.onsdigital.co.uk>
+   [grafana-dev-nomad-tasks]: <https://grafana.develop.onsdigital.co.uk/d/000000014/nomad-task?refresh=10s&orgId=1>
```
Comment (eldeal) [inline training/platform-services/PLATFORM.md:177]: I'm not certain these links will always work, the ID might change over time, but we can wait and see if it actually causes any problems because I really dont know. Might be worth asking in case anyone else knows though

File: training/platform-services/PLATFORM.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+   [grafana-dev-node-stats-2]: <https://grafana.develop.onsdigital.co.uk/d/000000004/node-stats-2?refresh=10s&orgId=1>
+   [prometheus-dev-home]: <https://prometheus.develop.onsdigital.co.uk>
+   [prometheus-dev-alerts]: <https://prometheus.develop.onsdigital.co.uk/alerts>
+   [prometheus-querying]: <https://prometheus.io/docs/prometheus/latest/querying/basics>
+   [support-101-slides]: <https://docs.google.com/presentation/d/1AHUouPbIbOwFzAVRnfVIJNLifxZm512kHJZBe5X5BwM/edit#slide=id.ga77d8de13d_0_161>
```
Comment (eldeal) [inline training/platform-services/PLATFORM.md:182]: Can we instead link to the module in the training directory for support please? :)

**Review by eldeal — CHANGES_REQUESTED** [review]

File: training/platform-services/PLATFORM.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+More information about Nomad can be found on their website - https://www.nomadproject.io/
+
+## Kibana
+Visit the following to access Kibana
+- DEVELOP:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;https://kibana.develop.onsdigital.co.uk
+- PRODUCTION:&nbsp;&nbsp;&nbsp;https://kibana.production.onsdigital.co.uk
+
+[<img src="images/kibana-dev-home.png" title="Kibana Develop Home" width=49.7% height=49.7%/>][kibana-dev-home] [<img src="images/kibana-dev-discover.png" title="Kibana Develop Discover" width=49.7% height=49.7%/>][kibana-dev-home]
+
+Kibana provides log aggeration where all logs are pushed to Kibana, allowing us to check the logs of all applications. With the use of logs in Kibana, this assists in debugging. We can filter the namespace to check logs of a specific application and filter with other properties to find information more quickly
```
[the remark this answers is not in the document]
Comment (justinpjose) [reply training/platform-services/PLATFORM.md]: Agreed and done

File: training/platform-services/PLATFORM.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+- DEVELOP:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;https://vault.develop.onsdigital.co.uk
+- PRODUCTION:&nbsp;&nbsp;&nbsp;https://vault.production.onsdigital.co.uk
+
+**To sign into Vault, you require a `token` which can be retrieved from the [helper script in dp-setup][helper-script-dp-setup]**
+
+[<img src="images/vault-dev-login.png" title="Vault Develop Login" width=49.7% height=49.7%/>][vault-develop] [<img src="images/vault-dev-home.png" title="Vault Develop Home" width=49.7% height=49.7%/>][vault-develop]
+
+> Vault secures, stores and tightly controls access to tokens, passwords, certificates, encryption keys for protecting secrets and other sensitive data
+
+Vault manage secrets and stores the secrets in [dp-configs][dp-configs-secrets] in a secure way and made available to the service (which can turn feature on and off). In general, Vault is used for the following two functionalities:
```
[the remark this answers is not in the document]
Comment (justinpjose) [reply training/platform-services/PLATFORM.md]: Agreed and done

File: training/platform-services/PLATFORM.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+1. Storing application configs (secrets) from [dp-configs][dp-configs-secrets]
+2. Storing per-file encryption keys for CMD uploads, image uploads, etc. 
+   
+Some apps that deal with S3 uploads (florence, dataset exporter etc) need a vault client in the app and hence this is informed in their Makefile) while other apps only retrieve configs. Please note that all apps need their [vault policies][vault-policy] defined in [dp-setup/ansible/files/vault-policies][dp-setup-vault-policies]. Policies provide a declarative way to grant or forbid access to certain paths and operations in Vault. That’s why if you look at the vault policies, most will only have one path while some (such as [florence vault policy][florence-vault-policy] which deals with S3 uploads) will have multiple.
```
[the remark this answers is not in the document]
Comment (justinpjose) [reply training/platform-services/PLATFORM.md]: Agreed and done

File: training/platform-services/PLATFORM.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+Run `dp remote allow develop` and `dp remote allow concourse` to ensure you can access concourse and the develop environment.
+Please note that if you have production access, you can run `dp remote allow production` to access production environment.
+
+[Cloud Technologies and Infrastructure](../services/INFRASTRUCTURE.md) - You should have an understanding of AWS and the infrastructure as the platform services are very closely related to this
+
+## Structure
+![](images/platform-structure.png)
+
+## Contents
```
[the remark this answers is not in the document]
Comment (justinpjose) [reply training/platform-services/PLATFORM.md:18]: Agreed and done

This is more structured on a high level as well which is nice. I feel there is no need to split up the content in this documentation. If we do, each documentation would contain little information. I think it would be better if everything is in one place. Also, the Contents section gives navigation to the documentation where a person can navigate to the specific service information quickly

If more documentation is made on specific services, then it is worth linking them to this documentation. This documentation informs mainly high-level detail of the services.

File: training/platform-services/PLATFORM.md (hunk 1 above)
Comment (justinpjose) [reply training/platform-services/PLATFORM.md]: Agreed and done

File: training/platform-services/PLATFORM.md (hunk 2 above)
Comment (justinpjose) [reply training/platform-services/PLATFORM.md]: Agreed and done

Comment (justinpjose) [reply training/platform-services/PLATFORM.md]: Yeah. That makes sense
Agreed and done

File: training/platform-services/PLATFORM.md (hunk 4 above)
Comment (justinpjose) [reply training/platform-services/PLATFORM.md]: Done

File: training/platform-services/PLATFORM.md (hunk 3 above)
Comment (justinpjose) [reply training/platform-services/PLATFORM.md:171]: I had a look online and I couldn't find any information regarding the change of ID (Grafana). But it is possible that the ID can change over time (as it is a logical thing that can happen). I believe that it is worthwhile to keep the links in the documentation to allow easy navigation to different locations. 

In the future, if there is a change in ID, then the best route is to remove the specific links. Either way, if this happens, we can learn from it that the ID can change and it would very minimal impact anyway (just link would direct to a 404)

**Review by DavidSubiros — APPROVED** [review]
I think the new documentation is very clear. I see that all of @eldeal comments have been addressed except the grafana link, but if we only have that link, then I think we should merge it as it is, and maybe change it in the future.

We could also create a new infrastructure task to have a well-defined URL that points to the right 'changing' URL (https://grafana.develop.onsdigital.co.uk/d/000000014/nomad-task?refresh=10s&orgId=1)?
